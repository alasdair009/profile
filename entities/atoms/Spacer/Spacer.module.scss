@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

.root {
  content: "";
  display: block;
  margin-bottom: calc(
    rem.toRem(map.get(dimensions.$sizes, s8)) * var(--multiplier)
  );
}
