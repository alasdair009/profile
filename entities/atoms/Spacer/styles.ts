import styled from "styled-components";
import {rem} from "polished";
import {sizes} from "../../design-tokens/dimensions";

export const Root = styled.br<{multiplier: number}>`
  content: "";
  display: block;
  margin-bottom: ${({ multiplier }) => rem(multiplier * sizes.s8.raw)};
`;