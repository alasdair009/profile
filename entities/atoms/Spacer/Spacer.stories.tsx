import {Spacer} from "./Spacer";
import {Meta, StoryObj} from "@storybook/react";

const meta: Meta<typeof Spacer> = {
    component: Spacer,
    args: {
        multiplier: 1
    }
}
export default meta;

export const Default: StoryObj<typeof Spacer> = {};