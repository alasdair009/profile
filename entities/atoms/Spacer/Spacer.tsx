import {Root} from "./styles";
import {HTMLAttributes, ReactNode} from "react";

type SpacerProps = {
    multiplier?: number;
} & HTMLAttributes<HTMLBRElement>;
export function Spacer({multiplier = 1, ...rest}: SpacerProps) {
    return <Root multiplier={multiplier} {...rest} />
}