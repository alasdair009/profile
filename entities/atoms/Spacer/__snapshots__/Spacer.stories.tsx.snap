// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Spacer Default play-test 1`] = `
<br data-testid="Spacer"
    class="sc-hDcvty llUJuV"
>
`;

exports[`atoms/Spacer Eight play-test 1`] = `
<br data-testid="Spacer"
    class="sc-hDcvty cnEEcO"
>
`;
